### ISSUE TITLE
Array is interpreted as property access

### ISSUE THREAD

**Issue by HoffmannP** [eslint/eslint#12786]
**Tell us about your environment**
standard 13.1.0
eslint 6.1.0
npm 6.13.4
node 10.7.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
see standard

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
export default function (el) {
  el.build()
  [1, 2].forEach(el.run)
}
```

```bash
standard minimal.js
```

**What did you expect to happen?**
no error

**What actually happened? Please include the actual, raw output from ESLint.**
```
standard: Use JavaScript Standard Style (https://standardjs.com)
standard: Run `standard --fix` to automatically fix some problems.
  minimal.js:3:1: Expected indentation of 4 spaces but found 2.
  minimal.js:3:3: Unexpected newline between object and [ of property access.
  minimal.js:3:5: Unexpected use of comma operator.
```

**Are you willing to submit a pull request to fix this bug?**
Willing but not capable

**Comment by g-plane** [comment]
Hello, please try adding a semicolon before your array at Line 3.

**Comment by aladdin-add** [comment]
yes, you can add a semi to avoid ASI unexpectly.

to understanding the ASI, you can read the docs: https://eslint.org/docs/rules/semi

**Comment by kaicataldo** [comment]
Just to clarify this, engines will not automatically insert a semicolon and so this will be interpreted as:

```js
export default function (el) {
  el.build()[1, 2].forEach(el.run);
}
```

Using a no-semicolon style, you'll have to write this as

```js
export default function (el) {
  el.build()
  ;[1, 2].forEach(el.run)
}
```

Since you're using Standard, that project's website has a [great description of this behavior](https://standardjs.com/rules.html#semicolons).
